### ISSUE TITLE
View is not refreshed when entering a folder after having drag-dropped images from the desktop onto that folder

### ISSUE THREAD

**Issue by oparoz** [owncloud/core#23200]
### Steps to reproduce

1. create a folder
2. drag a dozen image **from the desktop** and drop them on top of the new folder
3. Enter the folder

### Expected behaviour

The view should get updated as new images are uploaded

### Actual behaviour

I see the few images which have been uploaded before I entered the folder, but I don't get to see the rest until I refresh the page

### Server configuration

**PHP version:** 5.6

**ownCloud version:** stable9

**Comment by PVince81** [comment]
Yeah, this used to work, maybe the path recognition is broken (extra slashes or so).
Mind testing whether this also happens with https://github.com/owncloud/core/pull/21237 ?

**Comment by oparoz** [comment]
Yes.

**Comment by PVince81** [comment]
Possibly expansion for "Yes.":
1) Yes, it works with this PR
2) Yes, I'll test it

:wink:

**Comment by oparoz** [comment]
Yeah, sorry ;)

So, it's still failing, but so far, it's only been missing one file (that's maybe because it's happening very fast on localhost)

When dropping files on the breadcrumb, I always get all of them, but sometimes when entering the parent folder I get a blank page until the upload process is completed.

**Comment by PVince81** [comment]
It could also be a bit of a race condition. Maybe at the time the folder is being switched, the value of the current folder wasn't set yet, so when the event "file has finished uploaded" comes, the directory comparison still thinks we're in the old folder. Just guessing.

**Comment by oparoz** [comment]
Just also had the one missing file when dropping on the breadcrumb. I'm using 38 files now.
I also see the empty page sometimes when entering the folder and it refreshes at the end.

I would say the bug is fixed by the PR, because the view refreshes at the end of the process, but now someone needs to figure out why it's missing a file at times...

**Comment by PVince81** [comment]
Yeah, let's keep this ticket open to look into the race condition / add more race condition safety.

**Comment by oparoz** [comment]
I can't even read my own instructions properly... I was doing a MOVE... :tired_face: 

So doing a move I saw a new bug. Without leaving the current folder, 2 pictures got stuck in the current one after I dropped everything on the breadcrumb... Maybe that's the ones which don't make it to the other side at times.

Now on to this issue...

Yes, the PR fixes the issue. I switched to the new folder and saw the new ones being added one by one.
